src/blocknative.rs

```diff
@@ -0,0 +1,129 @@
+use super::{GasPriceEstimating, Transport};
+use anyhow::{anyhow, Context, Result};
+use serde::Deserialize;
+use std::time::Duration;
+
+// Gas price estimation with https://www.blocknative.com/gas-estimator , api https://docs.blocknative.com/gas-platform#example-request .
+
+const API_URI: &str = "https://api.blocknative.com/gasprices/blockprices";
+
+pub struct BlockNative<T> {
+    transport: T,
+    api_key: String,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct EstimatedPrice {
+    confidence: u64,
+    price: f64,
+    max_priority_fee_per_gas: f64,
+    max_fee_per_gas: f64,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct BlockPrice {
+    estimated_prices: Vec<EstimatedPrice>,
+}
+
+#[derive(Debug, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct Response {
+    block_prices: Vec<BlockPrice>,
+}
+
+impl<T: Transport> BlockNative<T> {
+    pub fn new(transport: T, api_key: String) -> Self {
+        Self { transport, api_key }
+    }
+
+    async fn gas_price(&self) -> Result<Response> {
+        self.transport
+            .get_json(API_URI, Some(self.api_key.clone()))
+            .await
+            .context("failed to get blocknative gas price")
+    }
+}
+
+#[async_trait::async_trait]
+impl<T: Transport> GasPriceEstimating for BlockNative<T> {
+    async fn estimate_with_limits(&self, _gas_limit: f64, _time_limit: Duration) -> Result<f64> {
+        let response = self.gas_price().await?;
+        if let Some(block) = response.block_prices.first() {
+            if let Some(estimated_price) = block.estimated_prices.first() {
+                return Ok(estimated_price.price);
+            }
+        }
```

Comment:
Their api docs  say
> Order of confidence intervals is subject to change.

So I think we should always sort the prices.

Reply:
True. Fixed.